lib/json_matchers/matcher.rb

```diff
@@ -10,16 +10,22 @@ def initialize(schema_path, **options)
     def matches?(response)
       @response = response
 
-      JSON::Validator.validate!(
-        schema_path.to_s,
-        response.body,
-        options,
-      )
-    rescue JSON::Schema::ValidationError => ex
-      @validation_failure_message = ex.message
-      false
-    rescue JSON::ParserError
-      raise InvalidSchemaError
+      begin
+        add_schemata_to_document_store
```

Comment:
Turns out, I had some issues with the `$ref` links of newly loaded schemas failing to expand references properly, leaving me with intermittent reference errors:

> #/definitions/relationships/properties/someResources: Couldn't resolve pointer "#/definitions/data". #: Couldn't resolve references: file:/anotherResource.json#/definitions/data.

After I went back to this approach of always loading the schemata again, I had no errors.

It might be that the approach of always re-reading schema files in this `#matches?` method isn't the best option. Maybe we could try querying the instantiated `JsonSchema::DocumentStore` instead to find the matching, loaded schema? This way it'd only be necessary to read the schema files once, and then use the schemas from memory. [`JsonSchema::DocumentStore#lookup_schema(uri)` method](https://github.com/brandur/json_schema/blob/v0.11.0/lib/json_schema/document_store.rb#L25) seems promising. What do you think?

It would mean that the public API would have to be changed to use the proper URI of the schema, but that might even be a good thing..?

```ruby
# spec
expect(response).to match_response_schema('file:/my_resource.json')
```

```js
// schema file
{
  "id": "file:/my_resource.json",
  // ...
}
```